### PR TITLE
FOUR-18603 cases started table and view the list of all cases in the system ep

### PR DESCRIPTION
**Pull request by devmiguelangel**
## Issue & Reproduction Steps
Cases started table and View the list of all cases in the system (EP)

## Solution
- Add `cases_started` migration
- Add `CaseStarted` model
- Add `get_all_cases` endpoint v1.1
- Add `CaseController` controller and `CaseResource` resource

## Related Tickets & Packages
[FOUR-18603](https://processmaker.atlassian.net/browse/FOUR-18603)

## Code Review Checklist
- [ ] I have pulled this code locally and tested it on my instance, along with any associated packages.
- [ ] This code adheres to [ProcessMaker Coding Guidelines](https://github.com/ProcessMaker/processmaker/wiki/Coding-Guidelines).
- [ ] This code includes a unit test or an E2E test that tests its functionality, or is covered by an existing test.
- [ ] This solution fixes the bug reported in the original ticket.
- [ ] This solution does not alter the expected output of a component in a way that would break existing Processes.
- [ ] This solution does not implement any breaking changes that would invalidate documentation or cause existing Processes to fail.
- [ ] This solution has been tested with enterprise packages that rely on its functionality and does not introduce bugs in those packages.
- [ ] This code does not duplicate functionality that already exists in the framework or in ProcessMaker.
- [ ] This ticket conforms to the PRD associated with this part of ProcessMaker.

ci:next

[FOUR-18603]: https://processmaker.atlassian.net/browse/FOUR-18603?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: ProcessMaker/Http/Controllers/Api/V1_1/CaseController.php [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+     *
+     * @return void
+     */
+    private function sortBy(Request $request, Builder $query): void
+    {
+        $sort = explode(',', $request->get('sortBy'));
+
+        foreach ($sort as $value) {
+            if (!preg_match('/^[a-zA-Z_]+:(asc|desc)$/', $value)) {
+                continue;
```
Comment (caleeli) [inline ProcessMaker/Http/Controllers/Api/V1_1/CaseController.php:134]: Please consider to return 422 if the sort has wrong parameters

File: ProcessMaker/Http/Controllers/Api/V1_1/CaseController.php [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+            }
+
+            $sort = explode(':', $value);
+            $field = $sort[0];
+            $order = $sort[1] ?? 'asc';
```
Comment (caleeli) [inline ProcessMaker/Http/Controllers/Api/V1_1/CaseController.php:139]: 'asc' could be a constant of the class

File: ProcessMaker/Http/Controllers/Api/V1_1/CaseController.php [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+     */
+    private function search(Request $request, Builder $query): void
+    {
+        if ($request->has('search')) {
+            $search = $request->get('search');
+
+            $query->where(function ($q) use ($search) {
+                foreach ($this->searchableFields as $field) {
+                    if ($field === 'case_number') {
+                        $q->orWhere($field, 'like', "%$search%");
```
Comment (caleeli) [inline ProcessMaker/Http/Controllers/Api/V1_1/CaseController.php:192]: Consider to change this to equal operation, and add that detail to the architecture document to search by case_number

File: ProcessMaker/Http/Requests/GetAllCasesRequest.php [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    /**
+     * Get the validation rules that apply to the request.
+     *
+     * @return array<string, \Illuminate\Contracts\Validation\ValidationRule|array<mixed>|string>
+     */
+    public function rules(): array
+    {
+        return [
+            'userId' => 'sometimes|integer',
+            'status' => 'sometimes|in:in_progress,completed',
```
Comment (caleeli) [inline ProcessMaker/Http/Requests/GetAllCasesRequest.php:26]: Please use uppercase for the status to keep the code style of PM, and also update it to the architecture document.

File: ProcessMaker/Http/Resources/V1_1/CaseResource.php [hunk 5]
```diff
@@ -0,0 +1,34 @@
+<?php
+
+namespace ProcessMaker\Http\Resources\V1_1;
+
+use ProcessMaker\Http\Resources\ApiResource;
+
+class CaseResource extends ApiResource
+{
+    protected static $defaultFields = [
```
Comment (caleeli) [inline ProcessMaker/Http/Resources/V1_1/CaseResource.php:9]: Add documentation to clarify this are the field that are going to be returned as response of the endpoint

File: ProcessMaker/Http/Controllers/Api/V1_1/CaseController.php [hunk 6]
```diff
@@ -0,0 +1,200 @@
...
+use Illuminate\Http\Request;
+use Illuminate\Http\Response;
+use ProcessMaker\Http\Controllers\Controller;
+use ProcessMaker\Http\Requests\GetAllCasesRequest;
+use ProcessMaker\Http\Resources\V1_1\CaseResource;
+use ProcessMaker\Models\CaseStarted;
+
+class CaseController extends Controller
+{
+    protected $defaultFields = [
```
Comment (caleeli) [inline ProcessMaker/Http/Controllers/Api/V1_1/CaseController.php:15]: Add documentation to clarify this are the field that are going to be selected if the main query of the endpoint.

File: ProcessMaker/Http/Controllers/Api/V1_1/CaseController.php (hunk 3 above)
Comment (devmiguelangel) [reply ProcessMaker/Http/Controllers/Api/V1_1/CaseController.php:192]: It was fixed

File: ProcessMaker/Http/Requests/GetAllCasesRequest.php (hunk 4 above)
Comment (devmiguelangel) [reply ProcessMaker/Http/Requests/GetAllCasesRequest.php:26]: It was fixed

File: ProcessMaker/Http/Controllers/Api/V1_1/CaseController.php (hunk 6 above)
Comment (devmiguelangel) [reply ProcessMaker/Http/Controllers/Api/V1_1/CaseController.php:15]: Documentation was added

**Pull request merged**